### PR TITLE
Add db module

### PR DESCRIPTION
**Pull request by pablomendezroyo**
Add db module

### CODE REVIEW

File: src/db/db.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import sqlite3 from "sqlite3";
+
+export interface Target {
```
Comment (dsimog01) [inline src/db/db.ts:3]: We should remember to place this in the proper place in the future, but ok for now

File: src/db/db.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * DatabaseHandler is a class that handles the database connection and
+ * provides methods for writing, reading and removing data from the database.
+ */
+export class DatabaseHandler {
+  private db: sqlite3.Database;
+
+  constructor(databaseName: string) {
+    this.db = new sqlite3.Database(databaseName, (err) => {
```
Comment (dsimog01) [inline src/db/db.ts:16]: Shouldn't there be some authentication here?

File: src/db/db.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      if (err) {
+        err.message += `. Failed connecting to database ${databaseName}. Error: ${err.message}`;
+        throw err;
+      }
+      console.log("Connected to the database.");
+      this.db.run(
+        `CREATE TABLE IF NOT EXISTS data(instance TEXT PRIMARY KEY, email TEXT)`,
```
Comment (dsimog01) [inline src/db/db.ts:23]: Remember there needs to be some identification for the monitor that will handle this target

File: src/db/db.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        }
+      });
+    });
+  }
+
+  /**
+   * Reads all targets from the database.
+   * @returns All targets from the database
+   */
+  async readTargets(): Promise<Target[]> {
```
Comment (dsimog01) [inline src/db/db.ts:97]: Maybe we could add another function to get targets by monitor

**Review by dsimog01 — CHANGES_REQUESTED** [review]

File: src/db/db.ts (hunk 2 above)
Comment (pablomendezroyo) [reply src/db/db.ts:16]: No since the database will be stored locally in the same docker container

File: src/db/db.ts (hunk 3 above)
Comment (pablomendezroyo) [reply src/db/db.ts:23]: Adding it!

File: src/db/db.ts (hunk 4 above)
Comment (pablomendezroyo) [reply src/db/db.ts]: Implemented

**Pull request merged**